include/deal.II/lac/la_parallel_vector.h

```diff
@@ -1310,6 +1330,11 @@ namespace LinearAlgebra
        */
       mutable std::mutex mutex;
 
+      /**
+       * Communicator to be used for the shared-memory domain.
+       */
+      mutable MPI_Comm comm_sm = MPI_COMM_SELF;
```

Comment:
Do not initialize at the place of in-class declaration.

Reply:
Follow-up question: Why does the communicator need to be mutable?

Reply:
> Do not initialize at the place of in-class declaration.

Reply:
I have made the changes.